kernel/cooking.ml

```diff
@@ -217,6 +202,18 @@ let lift_univs cb subst auctx0 =
     let subst, auctx = discharge_abstract_universe_context subst auctx0 auctx in
     subst, (Polymorphic auctx)
 
+let cook_constr { Opaqueproof.modlist ; abstract } c =
+  let cache = RefTable.create 13 in
+  let abstract, usubst, abs_ctx = abstract in
+  (* For now the STM only handles deferred computation of monomorphic
+    constants. The API will need to be adapted when it's not the case
+    anymore. *)
+  let () = assert (AUContext.is_empty abs_ctx) in
+  let expmod = expmod_constr_subst cache modlist usubst in
+  let hyps0 = Context.Named.map expmod abstract in
```

Comment:
Hyps0 used to be used later on in the original code you copied. Here, and also there given the other change, there is no point in keeping this value alive. I think you should call it hyps so that it is shadowed.

Reply:
Fair enough. I was trying to keep the code of `cook_constr` and `cook_constant` as close as possible to easily remember to factorize them when implementing the delayed type-checking of polymorphic constants, but it doesn't hurt messing a bit with it already.